workflow/executor/executor.go

```diff
@@ -608,13 +609,23 @@ func (we *WorkflowExecutor) SaveLogs(ctx context.Context) {
 		}
 	}
 
-	// try to upsert TaskResult, if it fails, we will try to update the Pod's Annotations
+	return logArtifacts
+}
+
+func (we *WorkflowExecutor) ReportOutputs(ctx context.Context, logArtifacts []wfv1.Artifact) {
 	err := we.reportOutputs(ctx, logArtifacts)
 	if err != nil {
 		we.AddError(err)
 	}
 }
 
+func (we *WorkflowExecutor) FinalizeOutput(ctx context.Context) {
+	err := we.finalizeOutput(ctx)
+	if err != nil {
+		we.AddError(err)
+	}
+}
```

Comment:
do we need to have 2 levels of methods: exposed and unexposed? I don't think the other methods here are doing that, right? I'd be interested in keeping it more concise and consistent.

Reply:
Will do.